src/message/fetchActions.js

```diff
@@ -172,16 +172,7 @@ const initialFetchCompletePlain = (): Action => ({
 });
 
 export const initialFetchComplete = () => async (dispatch: Dispatch, getState: GetState) => {
-  if (!getNavigationRoutes().some(navigationRoute => navigationRoute.name === 'main-tabs')) {
-    // If we're anywhere in the normal UI of the app, then remain
-    // where we are. Only reset the nav state if we're elsewhere,
-    // and in that case, go to the main screen.
-    //
-    // TODO: "elsewhere" is probably just a way of saying "on the
-    // loading screen", but we're not sure. We could adjust the
-    // conditional accordingly, if we found out we're not depending on
-    // the more general condition; see
-    //   https://github.com/zulip/zulip-mobile/pull/4274#discussion_r505941875
+  if (getNavigationRoutes().some(navigationRoute => navigationRoute.name === 'loading')) {
```

Comment:
Could this drop the `some`, and instead look just at the root route, i.e. `getNavigationRoutes()[0]`?

That would make the reasoning quoted in the commit message apply more directly -- it says

    * The possible roots of the nav stack are main-tabs, loading,
      account-pick, and realm-input. Those are the routes that we either
      reset to, or set as the initial route.
    
    * So we're talking about a possible change in behavior when the root
      route is account-pick or realm-input.

after concluding the old logic is equivalent to "root route is not main-tabs". Implicitly it's using that the new logic is equivalent to "root route is loading", in order to say that the remaining two possible root routes are the only cases where the behavior might change.

Reply:
Yep, sounds right!